contrib/elastic/go-elasticsearch.v5/elastictrace.go

```diff
@@ -0,0 +1,147 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+// Package elastic provides functions to trace the github.com/elastic/go-elasticsearch packages.
+package elastic // import "gopkg.in/DataDog/dd-trace-go.v1/contrib/elastic/go-elasticsearch
+
+import (
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math"
+	"net/http"
+	"regexp"
+	"strconv"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// NewRoundTripper returns a new http.Client which traces requests under the given service name.
+func NewRoundTripper(opts ...ClientOption) http.RoundTripper {
+	cfg := new(clientConfig)
+	defaults(cfg)
+	for _, fn := range opts {
+		fn(cfg)
+	}
+	return &roundTripper{config: *cfg}
+}
+
+// bodyCutoff specifies the maximum number of bytes that will be stored as a tag
+// value obtained from an HTTP request or response body.
+var bodyCutoff = 5 * 1024
+
+// TracedRoundTripper is a traced HTTP transport that captures Elasticsearch spans.
```

Comment:
The documentation here seems off 🙂

Reply:
This is still off. The type name is `roundTripper` not `TracedRoundTripper`.